### PR TITLE
Remove System.Net.Http package dependency

### LINKED ISSUE #36627: Libraries which target modern TFMs (.NETCoreApp, netstandard2.0, net461) bring in System.Net.Http package unnecessarily
Some of the libraries in this repository package reference System.Net.Http which then gets added as a dependency transitively in the consuming app. A package reference to System.Net.Http is undesireable as it brings in the entire netstandard1.x dependency graph and slows down restore. Additionally, potential future CVEs (and alerts surfaced by Component Governance) wouldn't impact libraries in this repository as the dependency would be avoided entirely.

The API exposed by System.Net.Http is already inbox in modern TFMs like .NETCoreApp, netstandard2.x and net461 and above and doesn't need to be brought in via a package reference.

If you accept a change in this domain, I would be happy to submit a PR. 

Related: https://github.com/dotnet/arcade/issues/13315

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/36627

Some of the libraries in this repository package reference System.Net.Http which then gets added as a dependency transitively in the consuming app. A package reference to System.Net.Http is undesireable as it brings in the entire netstandard1.x dependency graph and slows down restore. Additionally, potential future CVEs (and alerts surfaced by Component Governance) wouldn't impact libraries in this repository as the dependency would be avoided entirely.

The API exposed by System.Net.Http is already inbox in modern TFMs like .NETCoreApp, netstandard2.x and net471 and above and doesn't need to be brought in via a package reference. Adding a net472 TFM to Azure.Core avoids the dependency on recent .NET Framework targeted applications.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by jsquire** [comment]
FYI:  @annelo-msft, @JoshLove-msft, @christothes, @KrzysztofCwalina, @tg-msft, @AlexanderSher, @heaths

**For context:**
- [System.Net.Http issue history and discussion](https://github.com/dotnet/runtime/issues/20777#issuecomment-338418610)
- [.NET Framework System.Net.Http Fix](https://github.com/dotnet/corefx/pull/15659)

**Review by annelo-msft — COMMENTED** [review]
It looks like `using` statements may need to be updated in some files in order for the library to build correctly on net462.

**Comment by ViktorHofer** [comment]
I need to take a closer look (hadn't had time) but it's likely that the net462 TFM needs to continue using the `System.Net.Http` package but it would be good to add another .NET Framework TFM, i.e. `net472` that drops that package dependency.

**Comment by ViktorHofer** [comment]
This document is quite outdated, unfortunately. https://github.com/dotnet/sdk/issues/27857 tracks updating the existing documentation.

That said, I'm the owner of APICompat / PackageValidation and am happy to help you with the migration. From a technical standpoint, the following file lists all the options that the new APICompat exposes:
- https://github.com/dotnet/sdk/blob/5050efe9c74ce9fd5eea8e9e196e47c76ae73fab/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ApiCompat.ValidatePackage.targets#L26C25-L58

**Comment by ViktorHofer** [comment]
@jsquire the PR should be ready. Is anything missing or can this be merged in?

File: eng/Directory.Build.Common.targets
Comment (annelo-msft) [inline eng/Directory.Build.Common.targets:307]: Curious: if we were to upgrade our version of ApiCompat, would we be able to remove this line?

File: sdk/core/Azure.Core/src/Azure.Core.csproj
Comment (annelo-msft) [inline sdk/core/Azure.Core/src/Azure.Core.csproj:42]: In a comment, you said: 
> it would be good to add another .NET Framework TFM, i.e. net472 that drops that package dependency.

In general, we would like to avoid multi-targeting in Azure.Core, because of the complexity it adds, so it would be helpful for us to understand the need for this before merging this change.

Could you share more about why this is needed?

**Comment by jsquire** [comment]
> @jsquire the PR should be ready. Is anything missing or can this be merged in?

@ViktorHofer:  I'd like to hold one more team discussion on our end.  If no additional feedback comes out of it, then I think we're good to go.

File: eng/Directory.Build.Common.targets
Comment (ViktorHofer) [reply eng/Directory.Build.Common.targets:307]: Yes, the new APICompat handles this for you. It automatically finds the best matching contract from the baseline package.

File: sdk/core/Azure.Core/src/Azure.Core.csproj
Comment (ViktorHofer) [reply sdk/core/Azure.Core/src/Azure.Core.csproj:42]: Initially I thought that the net461 build can drop the System.Net.Http package dependency but I was wrong on that. The System.Net.Http package dependency is needed on .NET Framework < 4.7.1. Because of that, I introduced another .NET Framework TFM (net472 which is widely used in the stack) that drops the dependency.

This means that while you pay a little cost for building another inner build (net472), a consumer of Azure.Core doesn't transitively reference the System.Net.Http package anymore. Avoiding that package reference is important as System.Net.Http itself again brings in numerous other dependencies from the .NET Standard 1.x dependency graph.

**Comment by ViktorHofer** [comment]
@jsquire kind ping for a status update.

**Comment by annelo-msft** [comment]
@ViktorHofer, I am still working on this, and apologies for the delay.

File: sdk/core/Azure.Core/src/Azure.Core.csproj
Comment (annelo-msft) [inline sdk/core/Azure.Core/src/Azure.Core.csproj:43]: Can you share more about how including System.Net.Http as a reference in the net472 target prevents it from pulling in the transitive dependencies this target brings?

Comment (ViktorHofer) [reply sdk/core/Azure.Core/src/Azure.Core.csproj:43]: The PackageReference to `System.Net.Http` only exists for the net461 TFM but not for the net472 TFM as .NET Framework 4.7.2 includes the necessary API inbox (as part of the framework). The System.Net.Http package is undesirable as a (transitive) package reference as the package is quite old, not serviced anymore and brings in even more transitive dependencies that aren't serviced anymore.

Comment (annelo-msft) [reply sdk/core/Azure.Core/src/Azure.Core.csproj:43]: Thank you!  I had not understood the distinction between `<PackageReference/>` and `<Reference/>`.

**Review by annelo-msft — APPROVED** [review]

**Review by jsquire — APPROVED** [review]
Thanks for your patience, @ViktorHofer.

**Comment by ViktorHofer** [comment]
> Thanks for your patience, @ViktorHofer.

Sure. I don't have permissions to hit the merge button though.

**Comment by jsquire** [comment]
> > Thanks for your patience, @ViktorHofer.
> 
> Sure. I don't have permissions to hit the merge button though.

@annelo-msft:  Are we ready to merge or is there another perspective that we'd like to include in reviews?

**Pull request merged**
